server/src/main/scala/org/http4s/server/middleware/MaxActiveRequests.scala

```diff
@@ -9,70 +9,100 @@ package org.http4s.server.middleware
 import cats.syntax.all._
 import cats.data._
 import cats.effect._
-import cats.effect.concurrent.Semaphore
-import cats.effect.implicits._
 
 import org.http4s.Status
 import org.http4s.{Request, Response}
+import org.http4s.ContextRequest
 
 object MaxActiveRequests {
+
+  @deprecated(message = "Please use forHttpApp instead.", since = "0.21.14")
```

Comment:
We use this name in a lot of places, so if we're moving this direction, we should follow up and do it in more places.

Reply:
@rossabaker I actually like the other name just fine, but I needed a fresh name due to the binary compatibility change. I'm certainly open to suggestions on how to go about naming this.

Reply:
Oh, the `Concurrent => Sync`.  You mentioned that, and then I didn't spot it.  No, I think you did the right thing.